flask_meld/component.py

```diff
@@ -59,43 +58,54 @@ def load_module_from_path(full_path, module_name):
 
 
 class Component:
-    def __init__(self, id=None, request=None, **kwargs):
+    def __init__(self, id=None, form=None, **kwargs):
         if not id:
             id = uuid.uuid4()
+        self.errors = {}
+        self._form = None
         self.__dict__.update(**kwargs)
         self.id = id
-        self._form = None
-        self.errors = {}
 
         if hasattr(self, "form_class"):
             # tricky: https://flask-wtf.readthedocs.io/en/stable/api.html
             # need to pass formdata=None or flask-wtf will try to use the
             # flask request object to populate the form
             self._form = getattr(self, "form_class")(formdata=None)
-            self._set_form_data()
+            for field in self._form:
+                d = {"meld:model": field.name}
+                setattr(self._form[field.name], "render_kw", d)
+                if field.name in kwargs:
+                    self._set_field_data(field.name, kwargs[field.name])
+                    if field.name == "csrf_token":
+                        soup = BeautifulSoup(field.__call__(), features="html.parser")
+                        token = soup.find(attrs={"name": "csrf_token"}).get("value")
+                        self.csrf_token = token
+                else:
+                    setattr(self, field.name, None)
 
     def __repr__(self):
         return f"<meld.Component {self.__class__.__name__}-vars{self._attributes()})>"
 
     @property
     def _meld_attrs(self):
-        return ["id", "render", "validate"]
+        return ["id", "render", "validate", "updated"]
 
-    def _set_form_data(self, data=None):
-        if not data:
-            data = self._attributes()
-        form = self._form
-        for field in form.data:
-            if field in data:
-                setattr(form[field], "data", data[field])
+    def _set_field_data(self, field_name, value):
+        setattr(self._form[field_name], "data", value)
 
-    def validate(self):
+    def validate(self, field=None):
         if self._form:
-            validate = self._form.validate()
+            if field:
+                validate = field.validate(self._form)
+            else:
+                validate = self._form.validate()
             if not validate:
                 for field in self._form:
                     if field.errors:
                         self.errors[field.name] = field.errors
+                return False
```

Comment:
What if none of the `field`s in `self._form` have errors? Would this still be `False`?